src/hotspot/share/nmt/mallocTracker.cpp

```diff
@@ -239,13 +243,12 @@ bool MallocTracker::print_pointer_information(const void* p, outputStream* st) {
   const MallocHeader* likely_live_block = nullptr;
   {
     const size_t smallest_possible_alignment = sizeof(void*);
-    const uint8_t* here = align_down(addr, smallest_possible_alignment);
-    const uint8_t* const end = here - (0x1000 + sizeof(MallocHeader)); // stop searching after 4k
+    uintptr_t here = (uintptr_t)align_down(addr, smallest_possible_alignment);
+    uintptr_t end = MAX2(smallest_possible_alignment, here - (0x1000 + sizeof(MallocHeader))); // stop searching after 4k
```

Comment:
This works since we know that 0x1000 + sizeof(MallocHeader) < 16MB, which is where we bail out above. Otherwise, for here values < 1K+16, we would underflow and end up with a large `end` value.